code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/CarouselPushTemplate.kt

```diff
@@ -46,60 +44,51 @@ internal open class CarouselPushTemplate : AEPPushTemplate {
         val interactionUri: String?
     )
 
-    protected constructor(data: Map<String, String>) : super(data) {
-        carouselLayoutType = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CAROUSEL_LAYOUT, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CAROUSEL_LAYOUT}\" not found.")
-
-        rawCarouselItems = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.CAROUSEL_ITEMS, null
-        )
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CAROUSEL_ITEMS}\" not found.")
-
-        carouselOperationMode = DataReader.optString(
-            data,
-            PushTemplateConstants.PushPayloadKeys.CAROUSEL_OPERATION_MODE,
-            PushTemplateConstants.DefaultValues.AUTO_CAROUSEL_MODE
-        )
+    /**
+     * Initializes the push template with the given NotificationData.
+     *
+     * @param data the data to initialize the push template with
+     * @param fromIntent flag to denote if the push template was built from an intent
+     */
+    init {
```

Comment:
The constructor was made protected so it can only be accessed by inheriting `ManualCarouselPushTemplate` and `AutoCarouselPushTemplate`. The only way to create on object of this class from the other classes should be using `CarouselPushTemplate.createCarouselPushTemplate` so that the right carousel template object is returned. Could you please revert to using a protected constructor?

Reply:
Good call. I missed that

I did that by following code
internal open class CarouselPushTemplate protected constructor(data: NotificationData) : AEPPushTemplate(data)

Looks like kotlin is fun